client/src/main/java/com/flipkart/flux/client/runtime/FluxRuntimeConnectorHttpImpl.java

```diff
@@ -19,175 +19,83 @@
+     * Flux runtime API
+     *
+     * @see com.flipkart.flux.client.runtime.FluxRuntimeConnector#updateExecutionStatus(ExecutionUpdateData)
+     */
+    public void updateExecutionStatus(ExecutionUpdateData executionUpdateData) {
+       fluxHttpClient. postOverHttp(executionUpdateData,
+                     "/" + executionUpdateData.getStateMachineId() + "/" + executionUpdateData
+                         .getTaskId() + "/status");
+    }
 
-	/**
-	 * Interface method implementation. Increments the execution retries in persistence by invoking suitable Flux runtime API
-	 * @see com.flipkart.flux.client.runtime.FluxRuntimeConnector#incrementExecutionRetries(java.lang.Long, java.lang.Long)
-	 */
-	@Override
-	public void incrementExecutionRetries(Long stateMachineId,Long taskId) {
-		CloseableHttpResponse httpResponse = null;
-        httpResponse = postOverHttp(null,  "/" + stateMachineId + "/" + taskId + "/retries/inc");
-        HttpClientUtils.closeQuietly(httpResponse);
-	}
+    /**
+     * Interface method implementation. Increments the execution retries in persistence by invoking
+     * suitable Flux runtime API
+     *
+     * @see com.flipkart.flux.client.runtime.FluxRuntimeConnector#incrementExecutionRetries(java.lang.Long,
+     * java.lang.Long)
+     */
+    @Override
+    public void incrementExecutionRetries(Long stateMachineId, Long taskId) {
+        fluxHttpClient.postOverHttp(null, "/" + stateMachineId + "/" + taskId + "/retries/inc");
+    }
 
     /**
      * Interface method implementation. Posts to Flux Runtime API to redrive a task.
      */
     @Override
     public void redriveTask(Long taskId) {
-        CloseableHttpResponse httpResponse = null;
-        httpResponse = postOverHttp(null,  "/redrivetask/" + taskId);
-        HttpClientUtils.closeQuietly(httpResponse);
-    }
-	
-	/** Helper method to post data over Http */
-    private CloseableHttpResponse postOverHttp(Object dataToPost, String pathSuffix)  {
-        CloseableHttpResponse httpResponse = null;
-        HttpPost httpPostRequest;
-        httpPostRequest = new HttpPost(fluxEndpoint + pathSuffix);
-        try {
-            final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-            objectMapper.writeValue(byteArrayOutputStream, dataToPost);
-            httpPostRequest.setEntity(new ByteArrayEntity(byteArrayOutputStream.toByteArray(), ContentType.APPLICATION_JSON));
-            httpResponse = closeableHttpClient.execute(httpPostRequest);
-            final int statusCode = httpResponse.getStatusLine().getStatusCode();
-            if (statusCode >= Response.Status.OK.getStatusCode() && statusCode < Response.Status.MOVED_PERMANENTLY.getStatusCode() ) {
-                // all is well, TODO write a trace level log
-            } else {
-                // TODO: log status line here
-                throw new RuntimeCommunicationException("Did not receive a valid response from Flux core");
```

Comment:
Closing the connection here should solve the leak issue with having a thread cleanup expired and idle connections periodically.